gematria/datasets/pipelines/compile_modules_lib.py

```diff
@@ -106,8 +107,31 @@ def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
     )
 
 
+class ProcessAndFilterBBs(beam.DoFn):
+  """A Beam transform to process and filter BBs."""
+
+  def __init__(self, remove_memory_accessing_instructions: bool):
+    self._remove_memory_accessing_instructions = (
+        remove_memory_accessing_instructions
+    )
+
+  def setup(self):
+    self._bb_processor_filter = process_and_filter_bbs.BBProcessorFilter()
+
+  def process(self, bb_hex: str) -> Iterable[str]:
+    output_block = self._bb_processor_filter.remove_risky_instructions(
+        bb_hex, bb_hex, self._remove_memory_accessing_instructions
+    )
+    if output_block == '':
+      return []
+
+    return [output_block]
```

Comment:
IIRC, it is more common for `process()` functions to `yield` than return a collection.